### PR TITLE
implement group resource

### PR DESCRIPTION
**Pull request by chris-rock**
- implement group resource for linux, windows, mac and freebsd
- improve /etc/group parser
- extend test coverage

### CODE REVIEW

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

File: lib/resources/group.rb [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    @group = groupname.downcase
+    @domain = domain
+    @domain = @domain.downcase unless @domain.nil?
+
+    @cache = nil
+
+    # select group manager
+    @group_provider = nil
+    case vulcano.os[:family]
+    when 'ubuntu', 'debian', 'redhat', 'fedora', 'arch', 'darwin', 'freebsd'
```
Comment (arlimus) [inline lib/resources/group.rb]: vulano.os.unix?

Comment (chris-rock) [inline lib/resources/group.rb]: That is a great idea, #90 exposes the feature in our os resource

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

**Comment by arlimus** [comment]
This is a huge addition, thank you @chris-rock !!

**Comment by arlimus** [comment]
@delivery approve

**Comment by chef-delivery** [comment]
Change: 9f25444e-a40b-4d64-8c5c-627629b91d67 approved by: @arlimus

**Pull request merged**